### DISCUSSION TITLE
i18n example

### DISCUSSION THREAD

**Discussion by quick691fr** [refinedev/refine#6108] [Q&A] [answered]
Hi,

I downloaded the i18n-next example from the examples folder. I ran `yarn` and `yarn dev` to start the project, it's displaying the blog-posts list page. If I switch language to German and hit refresh button, I got this error :
![image](https://github.com/refinedev/refine/assets/375807/195ef32b-b621-4e93-a1d4-5bced0b064f7)

It seems translations are done on the client side and cause a Hydration problem ?

**Comment by BatuhanW** [comment]
Hey @quick691fr can you try to pass a value 'en', instead of undefined here and let me know if it fixes the issue?

https://github.com/refinedev/refine/blob/74fe3d024a73ec98a2043d25be1c0e4f8996ebe9/examples/i18n-nextjs/src/providers/i18n/index.ts#L18

**Reply by quick691fr** [reply]
Hi @BatuhanW,

If I do that, I don't have error anymore but language is forced as english if I refresh page instead of using cookie value.

**Reply by quick691fr** [reply]
@BatuhanW any news ?

**Reply by alicanerdurmaz** [reply]
Hello @quick691fr, I'm sorry you're having this issue.

The problem is that `i18next` can’t find the current language from the cookie on the server side in Next.js. I’ll check it out, but I need a bit of time.

Here’s a link with more details: [https://locize.com/blog/next-app-dir-i18n/](https://locize.com/blog/next-app-dir-i18n/)

**Comment by alicanerdurmaz** [comment, marked as answer]
Hello @quick691fr, we used https://github.com/amannn/next-intl package to support client and server translations, and hydration errors are gone. I'm closing the discussion now, feel free to open it again if you encounter any other problem 🙌

**Comment by quick691fr** [comment]
Thanks @alicanerdurmaz

**Reply by alicanerdurmaz** [reply]
I forgot to add the example link. Here it is: https://github.com/refinedev/refine/tree/main/examples/i18n-nextjs
